src/phpDocumentor/Compiler/Linker/Linker.php

```diff
@@ -141,6 +146,13 @@ public function substitute($item, ?DescriptorAbstract $container = null)
         return null;
     }
 
+    /**
+     * @param array<string|DescriptorAbstract|null>|(ArrayAccess<array-key,
```

Comment:
This param seems to be very complicated. This might be the cause of the failing smoke test. 

But hang in, since I'm working on a fix to solve this issue. Since it was already reported by a user. :-D

Reply:
:o I was totally expecting I broke the code, not a notation that was not read! Thanks for catching that!

It is a very complicated type indeed... I think a refactor of this class should use something like an ArrayObject and stick to that. Here we must pass an array or an iterable that is also ArrayAccess.

Reply:
This is also causing issues with CS:
https://github.com/phpDocumentor/phpDocumentor/pull/2251/checks?check_run_id=421826643#step:6:18
I don't know how they expect me to put all that in 120 chars and still put variable name after that :D

Reply:
I would agree with a `//phpcs:ignore` above this docblock :-D
The linker is one of the most complicated classes we have. So that's a pain anyhow.

Reply:
Ok. I tried `//phpcs:ignore` above the dockblock, I tried `//phpcs:ignore` the docblock and `phpcs:ignore` in the docblock too. Is there something I'm missing here?